docs/docs/guides/web3_plugin_guide/plugin_authors.md

```diff
@@ -210,3 +200,177 @@ public link(parentContext: Web3Context) {
+
+```typescript
+// custom_rpc_methods_plugin.ts
+import { Web3PluginBase } from 'web3-core';
+
+import { Web3Context } from './reexported_web3_context';
+
+export class CustomRpcMethodsPlugin extends Web3PluginBase {
+	public pluginNamespace = 'customRpcMethods';
+
+	public someMethod() {
+		return 'someValue';
+	}
+}
+
+// Module Augmentation
+declare module './reexported_web3_context' {
+	interface Web3Context {
+		customRpcMethods: CustomRpcMethodsPlugin;
+	}
+}
+
+export { Web3Context };
+```
+
+The following represent what your plugin users would see:
+
+![web3 context augmentation side effect](./assets/web3_context_augmentation_side_effect.png 'web3Context augmentation side effect')
+
+The above screenshot shows intellisense thinking `.customRpcMethods.someMethod` is available to call on the instance of `Web3`, even though we haven't registered `CustomRpcMethodsPlugin` - running this code would result in an error.
+:::
+
+### Re-exporting Web3Context
+
+Currently TypeScript's module augmentation only supports named exports, so the first step in augmenting `Web3Context` is to re-export it as a named export. To do this we're going to create a `reexported_web3_context.ts` file (the name of this file can be whatever, but for the sake of this guide, we're going to assume it's named `reexported_web3_context.ts` and is located within the same directory as our `custom_rpc_methods_plugin.ts` file). The file contents should be as follows:
+
+```typescript
+import { Web3Context } from 'web3-core';
+
+export { Web3Context };
+```
+
+### Re-declaring the Module
+
+Now we're going to tell TypeScript that we're interested in re-defining a module's (in this case `reexported_web3_context`) interface. In simpler terms, TypeScript is already aware of what methods and classes exist for each web3.js module, but when registering a plugin, we're adding additional methods and/or classes to the module's interface and TypeScript needs a little help understanding what's going to be available within the module after the plugin is registered.
+
+We start with the following:
+
+```typescript
+// custom_rpc_methods_plugin.ts
+import { Web3PluginBase } from 'web3-core';
+
+// Here we are importing our re-exported Web3Context
+// from the above section
+import { Web3Context } from './reexported_web3_context';
+
+export class CustomRpcMethodsPlugin extends Web3PluginBase {
+	public pluginNamespace = 'customRpcMethods';
+
+	public someMethod() {
+		return 'someValue';
+	}
+}
+
+// Here is where we are beginning to augment the Web3Context module
+declare module './reexported_web3_context' {...}
+```
+
+### Adding our Plugin's Interface
+
+Now that TypeScript's aware that the interface of the `reexport_web3_context` module is going to be augmented, we add our changes. In this case, we're adding the interface of `SimplePlugin` to the interface of `Web3Context` which is what we're going to be calling `.registerPlugin` on:
```

Comment:
Actually, "TypeScript`s" is not the same as "TypeScript is".  We can not abbreviate "is" with names. With names, the " 's " is used to show possession and ownership.

Also using you to keep with the context.

```suggestion
Now that TypeScript is aware that the interface of the `reexport_web3_context` module is going to be augmented, you can add our changes. In this case, you're adding the interface of `SimplePlugin` to the interface of `Web3Context` which is what the **plugin-user** is going to be calling `.registerPlugin` on:
```